operators/pkg/controller/elasticsearch/name/name.go

```diff
@@ -62,13 +64,28 @@ func suffix(name string, sfx string) string {
 
 // NewPodName returns a unique name to be used for the pod name and the
 // Elasticsearch cluster node name.
-func NewPodName(esName string) string {
-	sfx := stringsutil.Concat(
-		string(podSuffix),
-		"-",
-		rand.String(podRandomSuffixLength),
-	)
-	return suffix(esName, sfx)
+func NewPodName(esName string, nodeSpec v1alpha1.NodeSpec) string {
```

Comment:
looks like we can pass `nodeName string` directly instead of the entire `NodeSpec`?

Reply:
Yes that would be sufficient for now, but I figured it was rather light on context (two string vars, could be a little too easy to transpose), and wrapping in some other struct / type alias seemed a bit odd, so I went with accepting the container (`NodeSpec`) as a parameter.